src/modules/universal-discovery/components/finder/finder.tree.branch.component.js

```diff
@@ -10,17 +10,20 @@ export default class FinderTreeBranchComponent extends Component {
         super(props);
 
         this.expandBranch = this.expandBranch.bind(this);
+        this.handleClickLoadMore = this.handleClickLoadMore.bind(this);
+        this.renderLeaf = this.renderLeaf.bind(this);
+        this.updateSelectedLocations = this.updateSelectedLocations.bind(this);
 
         this.state = {
-            selectedLocations: props.selectedLocations,
-            currentlyLoadingLocationId: false
+            currentlyLoadingLocationId: null,
+            selectedLocationId: null,
         };
     }
 
-    componentWillReceiveProps(props) {
-        this.setState(state => Object.assign({}, state, {
-            selectedLocations: props.selectedLocations,
-            currentlyLoadingLocationId: false
+    UNSAFE_componentWillReceiveProps(props) {
```

Comment:
I think it would be good idea to start using new lifecycle methods, in this case `getDerivedStateFromProps`.